src/lib.rs

```diff
@@ -540,6 +540,83 @@ impl ByteSet {
         byte_set![byte]
     }
 
+    /// Construct a ByteSet from a `RangeTo` value, i.e. `..x`
+    #[inline]
+    #[must_use]
+    pub const fn from_range_to(range: ops::RangeTo<u8>) -> Self {
+        const fn chunk_for(this_chunk: usize, byte_chunk: usize, shift: usize) -> Chunk {
+            let value: Chunk = (1 << shift) - 1;
+            // the following code is equivalent to
+            // if this_chunk == byte_chunk {
+            //     value 
+            // } else if this_chunk < byte_chunk {
+            //     Chunk::max_value()
+            // } else {
+            //     0 
+            // }
+            //
+            // Once `if` works in const, this can be cleaned up
+            // https://github.com/rust-lang/rust/pull/72437
+            [
+                [
+                    0, Chunk::max_value()
+                ][(this_chunk < byte_chunk) as usize],
+                value
+            ][(this_chunk == byte_chunk) as usize]
```

Comment:
Please split the indexing out to two separate operations to make this easier to read.